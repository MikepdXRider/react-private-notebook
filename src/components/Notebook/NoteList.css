.noteList {
  padding: 0;
  width: 50%;
}

.noteListItem {
  list-style-type: none;
  padding: 0 10px;
}

.noteListItem:nth-child(2n) {
  background-color: var(--lighter-gray);
}

.note {
  display: grid;
  gap: 0.5rem;
  grid: 50px / 1.75fr 1fr;
  align-items: center;
}
