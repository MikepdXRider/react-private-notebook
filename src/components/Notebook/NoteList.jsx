import { Link } from 'react-router-dom';
import styles from './NoteList.css';

export default function NoteList({ notes, onDeleteNote }) {
  return (
    <ul aria-label="Notes" className={styles.noteList}>
      {notes.map((note) => (
        <li className={styles.noteListItem} key={note.id}>
          <article className={styles.note}>
            <h4>
              <Link to={`/notes/${note.id}`}>{note.title}</Link>
            </h4>
            <p>
              <Link to={`/notes/${note.id}/edit`}>
                <button aria-label={`Edit ${note.title}`}>Edit</button>
              </Link>{' '}
              <button
                aria-label={`Delete ${note.title}`}
                onClick={() => onDeleteNote(note)}
              >
                Delete
              </button>
            </p>
          </article>
        </li>
      ))}
    </ul>
  );
}
