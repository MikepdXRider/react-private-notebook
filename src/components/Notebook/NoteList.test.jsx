import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NoteList from './NoteList';

it('should display a list of notes', () => {
  const { container } = render(
    <MemoryRouter>
      <NoteList notes={[{ id: 1, title: 'Test note' }]} />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
  screen.getByText(/Test note/);
});

it('should call the delete function when clicked', () => {
  const mockDelete = jest.fn();
  render(
    <MemoryRouter>
      <NoteList
        notes={[{ id: 1, title: 'Test note' }]}
        onDeleteNote={mockDelete}
      />
    </MemoryRouter>
  );

  fireEvent.click(screen.getByRole('button', { name: 'Delete Test note' }));
});
