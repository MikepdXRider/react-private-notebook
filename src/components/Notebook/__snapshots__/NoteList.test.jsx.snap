// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a list of notes 1`] = `
<div>
  <ul
    aria-label="Notes"
    class="noteList"
  >
    <li
      class="noteListItem"
    >
      <article
        class="note"
      >
        <h4>
          <a
            href="/notes/1"
          >
            Test note
          </a>
        </h4>
        <p>
          <a
            href="/notes/1/edit"
          >
            <button
              aria-label="Edit Test note"
            >
              Edit
            </button>
          </a>
           
          <button
            aria-label="Delete Test note"
          >
            Delete
          </button>
        </p>
      </article>
    </li>
  </ul>
</div>
`;
